src/util/psbt/map/input.rs

```diff
@@ -150,11 +152,29 @@ pub struct Input {
 /// directly which signature hash type the user is dealing with. Therefore, the user is responsible
 /// for converting to/from [`PsbtSigHashType`] from/to the desired signature hash type they need.
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct PsbtSigHashType {
     pub (in ::util::psbt) inner: u32,
 }
+serde_string_impl!(PsbtSigHashType, "a PsbtSigHashType data");
+
+impl fmt::Display for PsbtSigHashType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.schnorr_hash_ty() {
+            Ok(schnorr_hash_ty) => fmt::Display::fmt(&schnorr_hash_ty, f),
+            Err(_) if self.inner <= 0xFF => write!(f, "{:#04x}", self.inner),
+            Err(_) => write!(f, "{:#010x}", self.inner),
```

Comment:
Perhaps just
```
        match self.schnorr_hash_ty() {
            Ok(schnorr_hash_ty) => fmt::Display::fmt(&schnorr_hash_ty, f),
            Err(_) => write!(f, "{:#x}", self.inner),
        }
```
Or was there some reason to pad with zeros for hex values greater than one byte?